product_variant_sale_order_route/models/sale_order_line.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    def product_id_change(self):
```

Comment:
This doesn't look right. IMO we need to redefine the domain applied to `route_id` dynamically according to the product.

Reply:
As we want a M2o on product.product, the onchange is right :wink: